### PR TITLE
ci: Add `git_ref` for E2E test runs on Head 

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Need to pass values in tags. Git_ref need to be defined
 
**How was this change tested?**
- N/A

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

File: .github/actions/e2e/setup-cluster/action.yaml [hunk 1]
```diff
@@ -82,6 +82,11 @@ runs:
       PRIVATE_CLUSTER: ${{ inputs.private_cluster }}
       GIT_REF:  ${{ inputs.git_ref }}
     run: |
+      if [[ "$GIT_REF" == '' ]]; then 
+        REF=$(git rev-parse HEAD)
+      else 
+        REF=$GIT_REF
+      fi
```
Comment (njtran) [inline .github/actions/e2e/setup-cluster/action.yaml:89]: Why can't we just use the same variable? 

```suggestion
      if [[ "$GIT_REF" == '' ]]; then 
        GIT_REF=$(git rev-parse HEAD)

```

File: .github/actions/e2e/setup-cluster/action.yaml [hunk 2]
```diff
@@ -188,12 +193,17 @@ runs:
...
       ACCOUNT_ID: ${{ inputs.account_id }}
       CLUSTER_NAME: ${{ inputs.cluster_name }}
-      GIT_REF:  ${{ inputs.git_ref }}
+      GIT_REF: ${{ inputs.git_ref }}
     run: |
+      if [[ "$GIT_REF" == '' ]]; then 
+        REF=$(git rev-parse HEAD)
+      else 
+        REF=$GIT_REF
+      fi
```
Comment (njtran) [inline .github/actions/e2e/setup-cluster/action.yaml:202]: Same here
```suggestion
      if [[ "$GIT_REF" == '' ]]; then 
        GIT_REF=$(git rev-parse HEAD)
```

**Review by jmdeal — APPROVED** [review]
LGTM 🚀

**Pull request merged**